lectures/sympy.md

```diff
@@ -0,0 +1,994 @@
+
+$$
+k^*:=\left(\frac{s A}{\delta}\right)^{1 /(1-\alpha)}
+$$ 
+
+This can be easily computed in SymPy
+
+```{code-cell} ipython3
+A, s, k, α, δ = symbols('A s k_t α δ')
+```
+
+```{code-cell} ipython3
+# Define Solow-Swan growth dynamics
+solow = Eq(s*A*k**α + (1-δ)*k, k)
+solow
+```
+
+```{code-cell} ipython3
+solve(solow, k)
+```
+
+### Inequalities and logic
+
+SymPy also allows users to define inequalities and set operators and provides a wide range of [operations](https://docs.sympy.org/latest/modules/solvers/inequalities.html).
+
+```{code-cell} ipython3
+reduce_inequalities([2*x + 5*y <= 30, 4*x + 2*y <= 20], [x])
+```
+
+```{code-cell} ipython3
+And(2*x + 5*y <= 30, x > 0)
+```
+
+### Series
+
+Series are widely used in economics and statistics, from asset pricing to the expectation of discrete random variables.
+
+We can construct a simple series of summations using `Sum` function and `Indexed` symbols
+
+```{code-cell} ipython3
+x, y, i, j = symbols("x y i j")
+sum_xy = Sum(Indexed('x', i)*Indexed('y', j), 
+            (i, 0, 3),
+            (j, 0, 3))
+sum_xy
+```
+
+To evaluate the sum, we can [`lambdify`](https://docs.sympy.org/latest/modules/utilities/lambdify.html#sympy.utilities.lambdify.lambdify) the formula.
+
+The lambdified expression can take numeric values as input for $x$ and $y$ and compute the result
+
+```{code-cell} ipython3
+sum_xy = lambdify([x, y], sum_xy)
+grid = np.arange(0, 4, 1)
+sum_xy(grid, grid)
+```
+
+#### Example: bank deposits
+
+Imagine a bank with $D_0$ as the deposit at time $t$.
+
+It loans $(1-r)$ of its deposits and keeps a fraction $r$ as cash reserves.
+
+Its deposits over an infinite time horizon can be written as
+
+$$
+\sum_{i=0}^\infty (1-r)^i D_0
+$$
+
+Let's compute the deposits at time $t$
+
+```{code-cell} ipython3
+D = symbols('D_0')
+r = Symbol('r', positive=True)
+Dt = Sum('(1 - r)^i * D_0', (i, 0, oo))
+Dt
+```
+
+We can call `doit` method to evaluate the series
+
+```{code-cell} ipython3
+Dt.doit()
+```
+
+Simplifying the expression above gives
+
+```{code-cell} ipython3
+simplify(Dt.doit())
+```
+
+This is consistent with the solution in the lecture on [geometric series](https://python.quantecon.org/geom_series.html#example-the-money-multiplier-in-fractional-reserve-banking).
+
+
+#### Example: discrete random variable
+
+In the following example, we compute the expectation of a discrete random variable.
+
+Let's define a discrete random variable $X$ following a [Poisson distribution](https://en.wikipedia.org/wiki/Poisson_distribution):
+
+$$
+f(x) = \frac{\lambda^x e^{-\lambda}}{x!}, \quad x = 0, 1, 2, \ldots
+$$
+
+```{code-cell} ipython3
+λ = symbols('lambda')
+
+# We refine the symbol x to positive integers
+x = Symbol('x', integer=True, positive=True)
+pmf = λ**x * exp(-λ) / factorial(x)
+pmf
+```
+
+We can verify if the distribution follows the fundamental property of probability distribution:
```

Comment:
@HumphreyYang @chappiewuzefan I am not sure what this means? What is the fundamental property?